enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,514 @@
+cluster. The second phase focuses on optimizing the resource
+utilization of the components included in the cluster and may further
+simplify the SNC image build process.
+
+### Phase One
+
+In the initial phase, the following annotation will be added to the
+manifest files for all components that are to be included in single
+node developer clusters:
+
+```
+include.release.openshift.io/single-node-developer=true
+```
+
+This annotation will need to be added to the manifests for all the
+OpenShift components that are not currently removed by the snc.sh
+script during the CRC build process. The components that are currently
+removed, and thus should not need the annotation added, include:
+
+* most operators running in the openshift-monitoring namespace
+* most operators running in the openshift-insights namespace
+* machine-api-operator
+* Machine-config-operator (may need to run once, requiring the annotation)
+* openshift-cloud-credential-operator
+* openshift-cluster-storage-operator
+
+Additional components that are not currently removed by the SNC
+tooling might be also be determined to be unnecessary, and thus not
+require the annotation. Possibilities include:
+
+* openshift-cluster-node-tuning-operator
+* openshift-multus
+
+Once PRs adding this annotation have been merged to all the necessary
+component repositories, the snc.sh script will be updated to tell the
+OpenShift installer to use the single-node-developer cluster profile,
+and to remove the code that deletes the components that are not
+annotated. At the conclusion of this initial phase, the somewhat
+simplified CRC build tooling should produce a cluster equivalent to
+that currently produced, and all existing CRC tests should pass. Phase
+one is expected to be completed during development of the 4.7
+OpenShift release.
+
+From this point forward, any new components being added to OpenShift
+will not be included in CRC unless a manifest with the
+single-node-developer annotation is included in the new component. The
+CRC team will need to engage with other OpenShift teams to help decide
+whether or not new components should be included in the profile, and
+if so, how they should be configured.
+
+### Phase Two
+
+In the second phase, various OpenShift components that are included in
+single node developer clusters will be updated to use specialized
+manifests in the single-node-developer profile rather than the same
+manifest used for default clusters. These specialized manifests will
+configure the components for use in single node developer clusters,
+replacing the patching currently done post-deployment by the CRC build
+process, and potentially enabling additional reductions in resource
+requirements.
+
+Some of the manifest specializations expected to be included in this
+phase are shown in the following table:
+
+
+| Operator | Change |
+| :--- | :--- |
+| ingress | reduce the number of replicas of controller pod |
```

Comment:
The changes described here would seem to be limited to manifests. What about changes that require modifying operator logic defined in code? 

That said, I'm not it makes sense to be exhaustively prescriptive as to what changes are required for each component. If component owners are going to be tasked with supporting resource-constrained deployment, an alternate strategy for ensuring success might be indicating what the requirements of that strategy are and leaving it to component owners to ensure those requirements are met. They wrote and maintain the operators, and are best equipped to implement the changes required provided requirements of sufficient clarity. Granted this kind of agency, I think they'd be able to contribute more effectively.